packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -519,15 +558,23 @@ export const transformPlotlyJsonToScatterChartProps = (
     return {
       legend,
       legendShape,
-      data: xValues.map((x, i: number) => ({
-        x: isString ? (isXDate ? new Date(x as string) : isXNumber ? parseFloat(x as string) : x) : x,
-        y: series.y[i],
-        ...(Array.isArray(series.marker?.size)
-          ? { markerSize: series.marker.size[i] }
-          : typeof series.marker?.size === 'number'
-          ? { markerSize: series.marker.size }
-          : {}),
-      })),
+      data: xValues
+        .map((x, i: number) => {
+          if (isInvalid(x) || isInvalid(series.y?.[i])) {
+            return null;
+          }
+
+          return {
+            x: isString ? (isXDate ? new Date(x as string) : isXNumber ? parseFloat(x as string) : x) : x,
+            y: series.y?.[i],
```

Comment:
>series.y?.[i] [](http://example.com/codeflow?start=15&length=13)

is this check needed as we are already returning null in such case above

Reply:
replaced ? with !